examples/lbfgs/driver.py

```diff
@@ -27,40 +27,37 @@
   # randomly initialized on each worker and would differ from worker to worker.
   # We pass the weights into the remote functions loss and grad so that the
   # weights are the same on each worker.
-  def net_initialization():
-    x = tf.placeholder(tf.float32, [None, image_dimension])
-    w = tf.Variable(tf.zeros(w_shape))
-    b = tf.Variable(tf.zeros(b_shape))
-    y = tf.nn.softmax(tf.matmul(x, w) + b)
-    y_ = tf.placeholder(tf.float32, [None, label_dimension])
-    cross_entropy = tf.reduce_mean(-tf.reduce_sum(y_ * tf.log(y), reduction_indices=[1]))
-    cross_entropy_grads = tf.gradients(cross_entropy, [w, b])
-
-    sess = tf.Session()
-
-    # In order to set the weights of the TensorFlow graph on a worker, we add
-    # assignment nodes. To get the network weights (as a list of numpy arrays)
-    # and to set the network weights (from a list of numpy arrays), use the
-    # methods get_weights and set_weights. This can be done from within a remote
-    # function or on the driver.
-    def get_and_set_weights_methods():
-      assignment_placeholders = []
-      assignment_nodes = []
-      for var in tf.trainable_variables():
-        assignment_placeholders.append(tf.placeholder(var.value().dtype, var.get_shape().as_list()))
-        assignment_nodes.append(var.assign(assignment_placeholders[-1]))
-
-      def get_weights():
-        return [v.eval(session=sess) for v in tf.trainable_variables()]
-
-      def set_weights(new_weights):
-        sess.run(assignment_nodes, feed_dict={p: w for p, w in zip(assignment_placeholders, new_weights)})
+  class LinearModel(object):
```

Comment:
Can you move this class definition before the __main__ section of the file (after the imports)? Same for net_initialization and net_reinitialization, loss, grad, full_loss and full_grad!